// api/resources/databaseinitialization/01-navbar/InsertNavbar.js
import Navbar from "../../../main/navbar/model/Navbar.js";

const insertNavbar = async () => {
  try {
    const navbars = [
      {
        name: "Root",
        icon: "",
        link: "/root",
        position: 1,
        parent_name: "",
        ui_group_name: "",
      },
      {
        name: "Dashboard",
        icon: "IoHomeOutline",
        link: "/",
        position: 1,
        parent_name: "Root",
        ui_group_name: "Root",
      },
      {
        name: "Admin",
        icon: "BsPeople",
        link: "/adminUser",
        position: 2,
        parent_name: "Root",
        ui_group_name: "Root",
      },
      {
        name: "Setup",
        icon: "BsMenuApp",
        link: "/setup",
        position: 3,
        parent_name: "Root",
        ui_group_name: "Root",
      },
      {
        name: "Faculty",
        icon: "PiChalkboardTeacher",
        link: "/faculty",
        position: 1,
        parent_name: "Setup",
        ui_group_name: "Setup",
      },
      {
        name: "Courses",
        icon: "LuBookMinus",
        link: "/course",
        position: 2,
        parent_name: "Setup",
        ui_group_name: "Setup",
      },
      {
        name: "Routine",
        icon: "MdMoreTime",
        link: "/routine",
        position: 3,
        parent_name: "Setup",
        ui_group_name: "Setup",
      },
      {
        name: "Notice",
        icon: "TfiAnnouncement",
        link: "/notice",
        position: 4,
        parent_name: "Setup",
        ui_group_name: "Setup",
      },
      {
        name: "Event",
        icon: "MdOutlineEventNote",
        link: "/event",
        position: 5,
        parent_name: "Setup",
        ui_group_name: "Setup",
      },
      {
        name: "News",
        icon: "FaRegNewspaper",
        link: "/news",
        position: 6,
        parent_name: "Setup",
        ui_group_name: "Setup",
      },
      {
        name: "Exam",
        icon: "PiExam",
        link: "/exam",
        position: 7,
        parent_name: "Setup",
        ui_group_name: "Setup",
      },
      {
        name: "Result",
        icon: "LuNotebookPen",
        link: "/result",
        position: 8,
        parent_name: "Setup",
        ui_group_name: "Setup",
      },
      {
        name: "Gallery",
        icon: "TfiGallery",
        link: "/gallery",
        position: 9,
        parent_name: "Setup",
        ui_group_name: "Setup",
      },
      {
        name: "Section",
        icon: "LuBookCheck",
        link: "/section",
        position: 10,
        parent_name: "Setup",
        ui_group_name: "Setup",
      },
      {
        name: "Student",
        icon: "PiStudent",
        link: "/student",
        position: 4,
        parent_name: "Root",
        ui_group_name: "Root",
      },
      {
        name: "Settings",
        icon: "IoSettingsOutline",
        link: "/settings",
        position: 5,
        parent_name: "Root",
        ui_group_name: "Root",
      },
      {
        name: "Site Management",
        icon: "GrSystem",
        link: "/siteManagement",
        position: 1,
        parent_name: "Settings",
        ui_group_name: "Settings",
      },
      {
        name: "About KCMIT",
        icon: "FaFileAlt",
        link: "/about",
        position: 1,
        parent_name: "Settings",
        ui_group_name: "Site_Management",
      },
      {
        name: "Mission",
        icon: "FaFileAlt",
        link: "/mission",
        position: 2,
        parent_name: "Settings",
        ui_group_name: "Site_Management",
      },
      {
        name: "Vision",
        icon: "FaFileAlt",
        link: "/vision",
        position: 3,
        parent_name: "Settings",
        ui_group_name: "Site_Management",
      },
      {
        name: "Principal Message",
        icon: "FaMessage",
        link: "/principalMessage",
        position: 4,
        parent_name: "Settings",
        ui_group_name: "Site_Management",
      },
      {
        name: "Director Message",
        icon: "FaMessage",
        link: "/directorMessage",
        position: 5,
        parent_name: "Settings",
        ui_group_name: "Site_Management",
      },
      {
        name: "Contact Us",
        icon: "RiContactsBook3Line",
        link: "/contact",
        position: 2,
        parent_name: "Settings",
        ui_group_name: "Settings",
      },
      {
        name: "Resources",
        icon: "PiDownloadSimpleBold",
        link: "/resources",
        position: 3,
        parent_name: "Settings",
        ui_group_name: "Settings",
      },
      {
        name: "Testimonials",
        icon: "TfiWrite",
        link: "/testimonials",
        position: 4,
        parent_name: "Settings",
        ui_group_name: "Settings",
      },
      {
        name: "System Configuration",
        icon: "TbSettingsCog",
        link: "/systemConfiguration",
        position: 5,
        parent_name: "Settings",
        ui_group_name: "Settings",
      },
      {
        name: "Event Category",
        icon: "MdEventAvailable",
        link: "/eventCategory",
        position: 1,
        parent_name: "Settings",
        ui_group_name: "System_Configuration",
      },
      {
        name: "Faculty Category",
        icon: "LuUserPen",
        link: "/facultyCategory",
        position: 2,
        parent_name: "Settings",
        ui_group_name: "System_Configuration",
      },
      {
        name: "Notice Category",
        icon: "MdOutlineNoteAlt",
        link: "/noticeCategory",
        position: 3,
        parent_name: "Settings",
        ui_group_name: "System_Configuration",
      },
      {
        name: "Notification Config",
        icon: "IoMdNotificationsOutline",
        link: "/notificationConfig",
        position: 6,
        parent_name: "Settings",
        ui_group_name: "Settings",
      },
    ];

    for (const navbar of navbars) {
      await Navbar.findOrCreate({
        where: { name: navbar.name, link: navbar.link },
        defaults: navbar,
      });
    }
    console.log("Navbar entries checked/created.");
  } catch (error) {
    console.error("Error in creating navbar entries:", error);
  }
};

export default insertNavbar;
